### PR TITLE
Add `region_isos` definitions for new scenarios

### LINKED ISSUE #241: update "r2dii.data::region_isos" to accommodate new scenarios
As we introduce new scenarios from new scenario providers we need new regional bridges. 

2 optiosn exist here iether separate files for region bridge specific to scenario input. 
eg "new_scenrio_input_file" must be used with "new_secanrio_region_file" 
or 
the already existing "r2dii.data::region_isos" accommodates for all scenarios. 

the latter is preferred as it reduces the number of files needed. 

So what we would need is to add new sources to "r2dii.data::region_isos" currently there is only "weo_2019" and "etp_2017" 

what we need now is 
region          isos                source 
example       example         weo_2019
                                           etp_2017 
                                           weo_2020 
                                           ieanze_2021
                                           nzaoa_2020 

exact names in source column to be given by @Antoine-Lalechere  - naming standard to be agreed upon in comments below 
@Antoine-Lalechere please provide the bridges here 

note that what ever is in r2dii.data::region_isos - column "source" muct = the column "scenario_source" in the scenario input file 

@jdhoffa @maurolepore @Antoine-Lalechere

### PR DESCRIPTION
**Pull request by jdhoffa**
Adds new regional data for the following scenarios: 
- WEO 2020
- ISF 2020
- NZE 2021

Thanks @Antoine-Lalechere 

Closes #241

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # r2dii.data (development version)
 
+* `region_isos` gains data for new scenario `sources`: `weo_2020`, `isf_2020` 
+and `nze_2021` (#241, @georgeharris2deg).
```
Comment (maurolepore) [inline NEWS.md:4]: ... new scenario `source`s instead?

File: data-raw/region_isos.R [hunk 2]
```diff
@@ -185,14 +181,62 @@ region_isos_etp_2017 <- region_data %>%
   warn_if_is_missing_country_isos() %>%
   rbind(global_data(., source_year)) %>%
   prepare_isos() %>%
-  bind_rows(ald_isos_etp_2017) %>%
+  bind_rows(mutate(ald_isos, source = source_year)) %>%
+  unique()
+
+# Process raw_regions_weo_2020.csv ----------------------------------------
+
+# Source: raw_regions_etp_2017.csv was transcribed from page 780 of the 2017
```
Comment (maurolepore) [inline data-raw/region_isos.R:189]: This comment is under a heading containing "2020" but refers to "2017". Is this intentional?

File: data-raw/region_isos.R [hunk 3]
```diff
@@ -185,14 +181,62 @@ region_isos_etp_2017 <- region_data %>%
...
+source_year <- "isf_2020"
+region_data <- read_regions(regions_path(source_year))
+
+region_isos_isf_2020 <- region_data %>%
+  pick_type("country_name") %>%
+  warn_if_is_missing_country_isos() %>%
+  rbind(global_data(., source_year)) %>%
+  prepare_isos() %>%
+  bind_rows(mutate(ald_isos, source = source_year)) %>%
+  unique()
```
Comment (maurolepore) [inline data-raw/region_isos.R:215]: DRY?

File: tests/testthat/test-region_isos.R [hunk 4]
```diff
@@ -5,3 +5,18 @@ test_that("hasn't changed", {
...
+    "weo_2020",
+    "isf_2020",
+    "nze_2021"
+  )
+
+  expect_equal(
+    sort(unique(region_isos$source)),
+    sort(expected_sources)
+  )
+})
```
Comment (maurolepore) [inline tests/testthat/test-region_isos.R:22]: :love_you_gesture:

**Review by maurolepore — COMMENTED** [review]
LGTM. My suggestion to DRY is mostly because I suspect we'll need to keep adding more scenario sources.

File: tests/testthat/test-region_isos.R (hunk 4 above)
Comment (jdhoffa) [reply tests/testthat/test-region_isos.R:22]: GO SURF

**Review by AlexAxthelm — APPROVED** [review]

**Pull request merged**